app/lib/task/backend.dart

```diff
@@ -329,26 +331,30 @@ class TaskBackend {
     String packageName, {
     bool updateDependants = false,
   }) async {
-    var lastVersionCreated = initialTimestamp;
-    await withRetryTransaction(_db, (tx) async {
-      final pkgKey = _db.emptyKey.append(Package, id: packageName);
+    final package = await packageBackend.lookupPackage(packageName);
+    if (package == null) {
+      return; // assume package was deleted!
+    }
+    final lastVersionCreated = package.lastVersionPublished ?? initialTimestamp;
+
+    // try to use cached version, purge cache and repopulate if it seems obsolete
+    var packageVersions = await packageBackend.listVersionsCached(packageName);
+    final lastCachedPublished =
+        packageVersions.versions.map((e) => e.published!).maxOrNull;
+    if (lastCachedPublished == null ||
+        lastCachedPublished.isBefore(lastVersionCreated)) {
+      await purgePackageCache(packageName);
+      packageVersions = await packageBackend.listVersionsCached(packageName);
+    }
+    // Determined the set of versions to track
+    final trackedVersions =
+        _versionsToTrack(package, packageVersions.versions).map(
+      (v) => v.canonicalizedVersion, // add extra sanity!
+    );
```

Comment:
If I'm in a race condition against another process running or publishing `trackPackage`, then I kind of have to reload the facts I'm using create my mutation inside `withRetryTransaction`.

If `withRetryTransaction` is retried due a conflicting mutation, then how do I know that the data I loaded before my transaction is still valid?


---------------------

Maybe, if `trackPackage` is called from `backfillTrackingState`, we could use cache values to check if mutations are necessary.
But:
 * Only when called from `backfillTrackingState`,
 * Only using cache value to conclude that an update isn't necessary.

We rarely have to do actual updates. I think consistency issues are a bit more concerning than extra compute spent once per week.